TST: organize and cleanup pandas/tests/groupby/test_aggregate.py

alysivji opened the pull request:
closes #18490

Module currently has tests in 2 classes (`TestGroupByAggregate` and `TestGroupByAggregateCython`). The remaining tests are not in a class.

Also made the following changes:

### TestGroupByAggregate class
#### test_agg_must_agg
* replaced `pytest.raises` with `tm.assert_raises_regex`

#### test_agg_apply_corner
* made it more readable

#### test_agg_multiple_functions_too_many_lambdas
* replaced `pytest.raises` with `tm.assert_raises_regex`

#### test_multi_function_flexible_mix
* made it more readable

### TestGroupByAggregateCython class

_all test methods that contain cython in the test name_

#### test_cython_agg_nothing_to_agg
* replaced `pytest.raises` with `tm.assert_raises_regex`

#### test_cython_agg_return_dict
* replaced `self.df` with df initialized inside function

### All Others

#### test_agg_dict_renaming_deprecation
* made it more readable

#### test_agg_nested_dicts
* replaced `pytest.raises` with `tm.assert_raises_regex`
* made it more readable

#### test_agg_item_by_item_raise_typeerror
* replaced `pytest.raises` with `tm.assert_raises_regex`

#### test_agg_structs_series
* made it more readable

alysivji committed:
Restructure tests

alysivji committed:
Change pytest.raises to pandas test helper

alysivji committed:
Replace pytest.raises with pandas test helper

alysivji committed:
Add vscode to gitignore

alysivji committed:
Fix failing test

jreback commented:
thanks @alysivji 

can you split things out to a new file (maybe just the moved tests)?

IOW each of these classes to a separate file, something like

``pandas/tests/groupby/aggregate/test_cython`` (2nd one)
``test_aggregate`` (1st class)
``test_other`` (3rd one)

just to make this even more organized.

be sure to add a ``__init__.py`` in the sub-dir

alysivji committed:
Split tests_aggregate.py tests into multiple files

alysivji committed:
Clean up imports

alysivji committed:
Change all assert_frame/assert_series to tm.assert_

alysivji committed:
Boyscouting -- cleaning up code

alysivji commented:
* Split up the files
* cleaned up the imports
* changed all assert_frame/assert_series calls to `tm.assert_*`
* more cleaning up tests to increase readability
* rebased

@jreback lemme know if anything else needs to be done.

Thanks!

jreback commented on pandas/tests/groupby/aggregate/test_aggregate.py:
```diff
@@ -0,0 +1,303 @@
+        expected = grouped.mean()
+        tm.assert_frame_equal(result, expected)
+
+    def test_aggregate_str_func(self):
```
can you parametrize this (on by_weekday and by_mwkday)

jreback commented on pandas/tests/groupby/aggregate/test_aggregate.py:
```diff
@@ -0,0 +1,303 @@
+        df['E'] = ['a'] * len(self.df)
+        grouped = self.df.groupby('A')
+
+        # API change in 0.11
```
can remove this commented out

jreback commented on pandas/tests/groupby/aggregate/test_cython.py:
```diff
@@ -0,0 +1,169 @@
+
+class TestGroupByAggregateCython(object):
+
+    def test_cythonized_aggers(self):
```
can you parameterize this

jreback commented on pandas/tests/groupby/aggregate/test_cython.py:
```diff
@@ -0,0 +1,169 @@
+        tm.assert_series_equal(summed, expected)
+
+    def test__cython_agg_general(self):
+        ops = [('mean', np.mean),
```
can you parameterize

jreback commented on pandas/tests/groupby/aggregate/test_cython.py:
```diff
@@ -0,0 +1,169 @@
+                exc.args += ('operation: %s' % op, )
+                raise
+
+    def test_cython_agg_empty_buckets(self):
```
same

jreback reviewed (changes requested):
@alysivji looking good. some requests to parameterize. could completely eliminate the need for the setup method in aggregate/test_aggregate.py if you want to make those into fixtures instead. can do this in another PR if you want.

jreback commented:
cc @TomAugspurger I think we have to merge #18921 (when ready) before this (and rebase this on master)?

TomAugspurger commented:
Yes it'd be nice to merge #18921 before this. Backporting can be tricky when files are split.

jreback commented:
@alysivji but go ahead and make the changes as above when you can. we will be merging @TomAugspurger soon anyhow.

alysivji committed:
Parametrize tests and remove commented out code

alysivji committed:
Delete cython class, all tests are grouped by module

alysivji commented:
@jreback Changes have been made. Also deleted the cython test class, not needed anymore since it's in a separate module.

I can rebase when @TomAugspurger 's changes get in.

Also started working on turning the `test_aggregate.py` setup into fixtures. Lots of moving pieces for one PR; will send a separate request once this is merged.

jreback commented on pandas/tests/groupby/aggregate/test_cython.py:
```diff
@@ -0,0 +1,180 @@
+        result = op(grouped)['C']
+        if name in ['sum', 'prod']:
+            tm.assert_series_equal(result, exp)
+
```
yes u can collapse testit - just move everything up

jreback commented on pandas/tests/groupby/aggregate/test_cython.py:
```diff
@@ -0,0 +1,180 @@
+    expected = df.groupby(labels).agg(targop)
+    try:
+        tm.assert_frame_equal(result, expected)
+    except BaseException as exc:
```
yes u can take this try/except out
it’s helpful when not parametrized but not now

jreback commented on pandas/tests/groupby/aggregate/test_aggregate.py:
```diff
@@ -0,0 +1,298 @@
+                                              ['C', grouped['C'].mean()],
+                                              ['D', grouped['D'].sem()]]))
+            tm.assert_frame_equal(result, expected)
+
```
same - collapse this function

jreback reviewed (changes requested):
looks good
couple of comments - will let u know when to rebase

alysivji committed:
Collapse functions and remove try except blocks

jreback commented:
@alysivji if you want to rebase we can get this in.

alysivji committed:
Orgnaize new tests

alysivji commented:
@jreback all green!

Will start working on refactoring the setup method to fixtures this weekend.

jreback commented:
thanks @alysivji 

would like to remove the imports *inside* functions as well (and consolidate at the top-level) on your next PR.
